src/server/_limiter.py

```diff
@@ -71,12 +71,13 @@ def get_multiples_count(request):
 def check_signals_allowlist(request):
     signals_allowlist = {":".join(ss_pair) for ss_pair in DashboardSignals().srcsig_list()}
     request_signals = []
-    if "signal" in request.args.keys():
-        request_signals += extract_strings("signal")
-    if "signals" in request.args.keys():
-        request_signals += extract_strings("signals")
-    if "data_source" in request.args:
-        request_signals = [f"{request.args['data_source']}:{request_signal}" for request_signal in request_signals]
+    request_args = request.args.keys()
+    if "signal" in request_args or "signals" in request_args:
+        source_signal_sets = parse_source_signal_sets()
+        for source_signal in source_signal_sets:
+            if isinstance(source_signal.signal, list):
+                for signal in source_signal.signal:
+                    request_signals.append(f"{source_signal.source}:{signal}")
     if len(request_signals) == 0:
         return False
     return all([signal in signals_allowlist for signal in request_signals])
```

Comment:
(not directly related to this change, and i dont know why we didnt think of this before, but...) could we make this a little more efficient by using a set for `request_signals` instead of a list?
```suggestion
    return request_signals.issubset(signals_allowlist)
```